week4/main.cpp

```diff
@@ -0,0 +1,73 @@
+//CodeU:
+//
+//  Created by Julia Semavina on 15.06.2017
+//
+//  problem specification:
+//      You are given a 2-dimensional map of tiles. Each tile is either land or water.
+//      You have to write a function that counts the number of islands.
+//
+//  comments:
+//      DFS
+//
+
+
+#include <iostream>
+#include <vector>
+
+
+void DFS(int x, int y, int IslandSize, std::vector<std::vector<bool>> &Island,
+         std::vector<std::vector<bool>> &Visited) {
+    Visited[y][x] = true;
+    if (x + 1 < IslandSize && Island[y][x + 1] && !Visited[y][x + 1]) { // right
+        DFS(x + 1, y, IslandSize, Island, Visited);
+    }
+    if (y + 1 < IslandSize && Island[y + 1][x] && !Visited[y + 1][x]) { // down
+        DFS(x, y + 1, IslandSize, Island, Visited);
+    }
+    if (x - 1 > -1 && Island[y][x - 1] && !Visited[y][x - 1]) { // left
+        DFS(x - 1, y, IslandSize, Island, Visited);
+    }
+    if (y - 1 > -1 && Island[y - 1][x] && !Visited[y - 1][x]) { // up
+        DFS(x, y - 1, IslandSize, Island, Visited);
+    }
+}
+
+
+int main() {
```

Comment:
Maybe reading the input and counting islands could be better split into functions?